tests/kernel/wave/wave_e2e_test.py

```diff
@@ -1623,3 +1624,60 @@ def test(
     test(a, scalar_c, scalar_d, b)
 
     assert torch.all(b == arg_vals[3]).item()
+
+
+@require_e2e
+@pytest.mark.parametrize("shape", [(3,64)])
+@param_bool("use_buffer_ops", "buf_ops")
+def test_atomic_min(shape, use_buffer_ops, request):
+    run_bench = request.config.getoption("--runperf")
+
+    M = tkl.sym.M
+    N = tkl.sym.N
+    ADDRESS_SPACE = tkl.sym.ADDRESS_SPACE
+
+    wave_size = 64
+    BLOCK_M = 1
+    BLOCK_N = sympy.Max(sympy.Min(N, 256), wave_size)
+
+    constraints: list[tkw.Constraint] = [
+        tkw.HardwareConstraint(
+            threads_per_wave=wave_size,
+            waves_per_block=(1, 1, 1),
+            vector_shapes={M: BLOCK_M, N: BLOCK_N},
+        )
+    ]
+    constraints += [tkw.WorkgroupConstraint(M, BLOCK_M, 1)]
+    constraints += [tkw.WorkgroupConstraint(N, BLOCK_N, 0)]
+    constraints += [tkw.WaveConstraint(M, BLOCK_M)]
+    constraints += [tkw.WaveConstraint(N, BLOCK_N)]
+
+    @tkw.wave(constraints)
+    def test(
+        a: tkl.Memory[M, N, ADDRESS_SPACE, tkl.i32],
+    ):
+        res = tkw.read(a, elements_per_thread=4)
+        shmem = tkw.allocate(shape=(M,N), distributed_shape=(BLOCK_M, BLOCK_N), dtype=tkl.i32, address_space=tkl.AddressSpace.SHARED_MEMORY)
+        zero_reg = tkl.Register[M, N, tkl.i32](55555)
```

Comment:
let's put this as 1e6 and call it inf_reg or something since it is not zeros

Reply:
done